src/Components/Components/src/Auth/AuthorizeView.razor

```diff
@@ -0,0 +1,68 @@
+@namespace Microsoft.AspNetCore.Components
+
+@if (currentAuthState == null)
+{
+    @Authorizing
+}
+else if (IsAuthorized())
+{
+    @((Authorized ?? ChildContent)?.Invoke(currentAuthState))
+}
+else
+{
+    @NotAuthorized
+}
+
+@functions {
+    private IAuthenticationState currentAuthState;
+
+    [CascadingParameter] public Task<IAuthenticationState> AuthStateTask { get; private set; }
```

Comment:
We should write out `Authentication` in all of this places. I'd also prefer if we can drop `Task` from the name.

Reply:
Actually is there a reason why this is public?

Reply:
Renamed.

Good point about visibility. It was just following the pattern of other params here, but you're right - the tooling limitation doesn't apply to CascadingParameter. Fixed.